### PR TITLE
CIRC-1726 Bundle multiple overdue fines into one single notice

### PR DESCRIPTION
**Pull request by alexanderkurash**
Resolves CIRC-1726

## Purpose
Bundle multiple fees/fines into a single notice for a single check-in session

## Approach
Add sessionId

### CODE REVIEW

File: descriptors/ModuleDescriptor-template.json [hunk 1]
```diff
@@ -774,6 +774,40 @@
           "unit": "minute",
           "delay": "2"
         },
+        {
+          "methods": [
+            "POST"
```
Comment (alexanderkurash) [inline descriptors/ModuleDescriptor-template.json:779]: Interface version

File: src/main/java/org/folio/circulation/resources/OverdueFineScheduledNoticeProcessingResource.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+import org.folio.circulation.infrastructure.storage.loans.LoanRepository;
+import org.folio.circulation.infrastructure.storage.notices.ScheduledNoticesRepository;
+import org.folio.circulation.infrastructure.storage.sessions.PatronActionSessionRepository;
+import org.folio.circulation.support.Clients;
+import org.folio.circulation.support.http.client.PageLimit;
+import org.folio.circulation.support.results.Result;
+
+import io.vertx.core.http.HttpClient;
+
+public class OverdueFineScheduledNoticeProcessingResource
```
Comment (alexanderkurash) [inline src/main/java/org/folio/circulation/resources/OverdueFineScheduledNoticeProcessingResource.java:32]: Logging

File: src/test/java/api/loans/OverdueFineScheduledNoticesProcessingTests.java [hunk 3]
```diff
@@ -82,6 +84,7 @@ void uponAtNoticeIsSentAndDeleted(TriggeringEvent triggeringEvent) {
     assertThatScheduledNoticeExists(triggeringEvent, UPON_AT, false, actionDateTime);
 
     scheduledNoticeProcessingClient.runFeeFineNoticesProcessing(rightAfter(actionDateTime));
```
Comment (alexanderkurash) [inline src/test/java/api/loans/OverdueFineScheduledNoticesProcessingTests.java:86]: Do we still need this processing to be triggered here?

File: src/test/java/api/loans/OverdueFineScheduledNoticesProcessingTests.java [hunk 4]
```diff
@@ -448,9 +466,14 @@ private void assertThatScheduledNoticeExists(TriggeringEvent triggeringEvent, No
   private void assertThatNoticesWereSent(UUID... expectedTemplateIds) {
     List<JsonObject> sentNotices = FakeModNotify.getSentPatronNotices();
 
-    assertThat(sentNotices, hasSize(expectedTemplateIds.length));
+    long numberOfCharges = sentNotices.stream()
+      .map(notice -> notice.getJsonObject("context").getJsonArray("feeCharges"))
+      .map(JsonArray::size)
+      .reduce(0, Integer::sum);
```
Comment (alexanderkurash) [inline src/test/java/api/loans/OverdueFineScheduledNoticesProcessingTests.java:472]: Add more tests where notices are actually bundled, make sure renewals are not bundled with check-ins, as well as check-ins from different sessions are not mixed

File: descriptors/ModuleDescriptor-template.json (hunk 1 above)
Comment (OleksandrVidinieiev) [reply descriptors/ModuleDescriptor-template.json:779]: It is a `_timer` interface, I'm not sure if we should update its version

File: src/test/java/api/loans/OverdueFineScheduledNoticesProcessingTests.java (hunk 3 above)
Comment (OleksandrVidinieiev) [reply src/test/java/api/loans/OverdueFineScheduledNoticesProcessingTests.java]: no, we don't

**Pull request merged**